src/java-interop/java-interop.targets

```diff
@@ -39,58 +24,64 @@
     <ClCompile Include="*.cc" />
   </ItemGroup>
 
-  <Target Name="_BuildJni_c"
-      Inputs="$(_JNIEnvGenPath)"
-      Outputs="$(IntermediateOutputPath)jni.c">
-    <MakeDir Directories="$(OutputPath)" />
-    <Exec Command="$(_RunJNIEnvGen) $(IntermediateOutputPath)jni.g.cs $(IntermediateOutputPath)jni.c" />
-  </Target>
+  <Target Name="_GetJavaInteropBuildInfo">
+    <PropertyGroup Condition=" $([MSBuild]::IsOSPlatform ('windows')) ">
+      <_PrepareArch   Condition=" '$(NETCoreSdkRuntimeIdentifier)' == 'win-x64' ">x86_amd64</_PrepareArch>
+    </PropertyGroup>
 
-  <Target Name="_GetCmakeOptions">
-    <PropertyGroup Condition=" '$(TargetFramework)' == 'net472' And '@(MonoIncludePath->Count())' != '0' ">
-      <_MonoDirs>"-DMONO_INCLUDE_LIST=@(MonoIncludePath, ';')"</_MonoDirs>
-      <_MonoLib>"-DMONO_LINK_FLAGS=$(MonoLibs)"</_MonoLib>
-      <_EnableMono>-DENABLE_MONO_INTEGRATION=ON</_EnableMono>
+    <PropertyGroup Condition=" $([MSBuild]::IsOSPlatform ('osx')) ">
+      <_CmakeOsxArch  Condition=" '$(NETCoreSdkRuntimeIdentifier)' == 'osx-x64'  ">x86_64</_CmakeOsxArch>
+      <_CmakeOsxArch  Condition=" '$(NETCoreSdkRuntimeIdentifier)' == 'osx-arm64'  ">arm64</_CmakeOsxArch>
     </PropertyGroup>
```

Comment:
Is it worth making it error on linux? Or can we guess what the paths might be there?

Reply:
These are values provided to the Cmake `CMAKE_OSX_ARCHITECTURES` variable.  `$(CmakeOsxArch)` is not used for Linux, and would make no sense there.

Reply:
Fair, but *currently* I don't *think* anything is missing for Linux.

(*Note*: I haven't actually built Linux yet, so Linux is basically a thought experiment until someone does so. 😉)